### ISSUE TITLE
Modules with path still can't be formatted on Windows

### ISSUE THREAD

**Issue by 95th** [rust-lang/rustfmt#3688]
#1754 issue is still occuring.

I tried `cargo fmt` on my tokio checkout and I get several errors like following:

```
error: couldn't read \\?\D:\tokio\tokio-sync\tests\..\src\mpsc\list.rs: The filename, directory name, or volume label syntax is incorrect. (os error 123)
  --> \\?\D:\tokio\tokio-sync\tests\fuzz_list.rs:14:5
   |
14 | mod list;
   |     ^^^^
```

code for above is:

```
#[path = "../src/mpsc/list.rs"]
#[allow(warnings)]
mod list;
```

**Comment by scampi** [comment]
Can you try the changes in https://github.com/rust-lang/rustfmt/pull/3590 and see if that fixes your problem ? Looks like it doesn't for some users...

**Comment by 95th** [comment]
It is not building:
```
   --> /home/95th/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_target-407.0.0/abi/mod.rs:112:62
    |
112 |                 ["a", ref a..] => dl.aggregate_align = align(a, "a")?,
    |                                                              ^ expected slice, found &str
    |
    = note: expected type `&[&str]`
               found type `&&str`

error[E0308]: mismatched types
   --> /home/95th/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_target-407.0.0/abi/mod.rs:113:58
    |
113 |                 ["f32", ref a..] => dl.f32_align = align(a, "f32")?,
    |                                                          ^ expected slice, found &str
    |
    = note: expected type `&[&str]`
               found type `&&str`

error[E0308]: mismatched types
   --> /home/95th/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_target-407.0.0/abi/mod.rs:114:58
    |
114 |                 ["f64", ref a..] => dl.f64_align = align(a, "f64")?,
    |                                                          ^ expected slice, found &str
    |
    = note: expected type `&[&str]`
               found type `&&str`

error[E0308]: mismatched types
   --> /home/95th/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_target-407.0.0/abi/mod.rs:117:46
    |
117 |                     dl.pointer_align = align(a, p)?;
    |                                              ^ expected slice, found &str
    |
    = note: expected type `&[&str]`
               found type `&&str`

error[E0308]: mismatched types
   --> /home/95th/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_target-407.0.0/abi/mod.rs:127:35
    |
127 |                     let a = align(a, s)?;
    |                                   ^ expected slice, found &str
    |
    = note: expected type `&[&str]`
               found type `&&str`

error[E0308]: mismatched types
   --> /home/95th/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_target-407.0.0/abi/mod.rs:145:35
    |
145 |                     let a = align(a, s)?;
    |                                   ^ expected slice, found &str
    |
    = note: expected type `&[&str]`
               found type `&&str`

error: aborting due to 13 previous errors

```

**Comment by 95th** [comment]
Hi @scampi, is there any update on this issue please?

**Comment by LPGhatguy** [comment]
I just hit this issue today in my repository. The repository and commit that is failing to rustfmt is:

https://github.com/rojo-rbx/rbx-dom/tree/3448e528766b1558ac76addf5e636df797f180ed

**Comment by scampi** [comment]
@95th Sorry for the delay, I had little time available for this. The branch is now compiling. Thanks for giving it a try!

**Comment by topecongiro** [comment]
This will be fixed in the 2.0 release.
